### PR TITLE
[CARBONDATA-224]Fixed data mismatch issue in case of Dictionary Exclude column for Numeric data type

### PR DESCRIPTION
**Pull request by kumarvishal09**
Problem: In case of greater than query on dictionary exclude column of for numeric data type
This is because data is sorted based on string because of this if data is 1,10,2,3 , data will be sorted like 1,10,2,3 but if we search greater than 3 then while applying min max will return false as last value is 3
Solution:we need to sort based on actual data type for this we should have chain comparator based on data type while loading the data, currently disabling DictionaryExclude column for numeric data type and will throw exception. Will raise jira issue to for sorting the based on actual data type

### CODE REVIEW

**Comment by kumarvishal09** [comment]
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/139/

**Comment by kumarvishal09** [comment]
raise Jira for handling data type based support https://issues.apache.org/jira/browse/CARBONDATA-225

**Comment by ravipesala** [comment]
@kumarvishal09 Please add testcase

File: integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala [hunk 1]
```diff
@@ -864,8 +864,8 @@ class CarbonSqlParser()
    /**
     * detects whether double or decimal column is part of dictionary_exclude
     */
-  def isDoubleDecimalColDictionaryExclude(columnDataType: String): Boolean = {
-    val dataTypes = Array("double", "decimal")
+  def isNumberInColDictionaryExclude(columnDataType: String): Boolean = {
+     val dataTypes = Array("double", "decimal", "int")
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: Add a reverse check that only string or (varchar-check) is only supported. As other datatypes like bigint etc will have problem in above check

Comment (kumarvishal09) [inline integration/spark/src/main/scala/org/apache/spark/sql/CarbonSqlParser.scala]: ok

**Comment by kumarvishal09** [comment]
@ravipesala Added testcase http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/195/

**Comment by gvramana** [comment]
LGTM, merged to master, @kumarvishal09 Raise separate PR for branch_0.1 also

**Pull request closed without merging**